CMakeLists.txt

```diff
@@ -132,7 +132,7 @@ add_custom_command(
 # Java
 find_package(Java REQUIRED COMPONENTS Runtime)
 set(ANTLR_JAR_LOCATION
-    ${PROJECT_SOURCE_DIR}/third_party/antlr4/antlr-4.9.1-complete.jar)
+    ${PROJECT_SOURCE_DIR}/third_party/antlr4/antlr4-4.9.3-complete.jar)
```

Comment:
If the files that are submitted to the repository in `src/parser/*` (for people who don't want java to be installed in the build process), these generated files should probably also be updated

(though can be a separate pull request)

(we could also should think about if we actually want to ship the pre-generated parser files or build them as part of the process always, that avoids freshness issues (e.g. weird timestamps of the checked out *.g4 vs. *.cpp files), and one can argue if someone compiles Surelog, they are also happy to install everything needed with it)

Reply:
I personally believe that developers should be willing to run the entire process rather than making auto-generated files part of the source deploy. It tends to hide the dependencies for not much gain. Moreover, Java setup is trivial regardless of platform and its only JRE not even the JDK requirement. At some point, we should really consider moving these auto-generated files out of the source folder and into the build directory.

Java is sort of a necessity for developers these days. If anything, Tcl is more of a pain on windows.

Reply:
I agree, so we should consider removing the files in `parser/`. @alaindargelas WDYT ?

Reply:
Another reason to either remove or move these file into the build folder is that Windows support multiple configurations and lot of our developers tend to run both debug & release builds in parallel which isn't possible with files being in source folders.

Similar problem exist in UHDM project as well. Both headers and src folders are in project's home directory. Those need to be moved into the build directory as well.

Reply:
Fully agree. The source folders should be considered readonly for the purpose of the build; all generated artifacts should end up in `build/` somewhere. Including code generated by the tcl scripts.